trieste/models/model_interfaces.py

```diff
@@ -37,7 +37,7 @@ def update(self, dataset: Dataset) -> None:
         raise NotImplementedError
 
     @abstractmethod
-    def predict(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, TensorType]:
+    def predict(self, query_points: QueryPoints, full_cov: bool = False) -> Tuple[ObserverEvaluations, TensorType]:
```

Comment:
as discussed, changing the structure of function return values based on arguments isn't recommended. I'll continue to think about it, but for posterity, the other options are

- have a completely separate method on `ProbabilsiticModel`, `predict_full_cov`, or some other name
- put the above method on a subclass of `ProbabilsiticModel`, but this will require us to make the model type generic across the code base. There are other cases where that might would help, but sth about it feels wrong
- have `predict` return a Distribution type, which can have the cov functionality on that

Reply:
I think for now that my favorite solution is predict_full_cov